agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -127,6 +130,9 @@ func TestTaskCleanupDoesNotDeadlock(t *testing.T) {
 		dockerId, err := agent.ResolveTaskDockerID(testTask, "1")
 		require.NoError(t, err, fmt.Sprintf("Cycle %d: Error resolving docker id for container in task", i))
 
+		err = testTask.Stop()
```

Comment:
The original deadlock that got introduced involved short-running containers that stopped themselves.  I think that by making this change you're no longer exercising the codepath that was deadlocking.